tests/CMakeLists.txt

```diff
@@ -29,5 +29,7 @@ if (CONFIG_ZTEST AND
   endif()
 endif()
 
+zephyr_include_directories(include)
 
 add_subdirectory_ifdef(CONFIG_UNITY	unity)
+add_subdirectory(mock_custom)
```

Comment:
I'm adding tests/include and tests/mock_custom directory in this PR. I'd like somebody to review that this is the right way to go.
@tejlmand Is it you or who should comment this?

Reply:
> I'm not sure I like the word `custom` in this case. Custom compared to what ? It's not like we already have some existing mocks, and then you customizing those for a specific purpose.
> 
> If I understand this correctly, then those are hand written mocks, as opposed to generated mocks.
> 
> So I think they should just be mocks, as I don't expect us to commit generated mocks to git. Generated mocks ought to go to the build folder, and preferable under a `generated` folder, so there should be no confusion in using just the name `mocks`.

Just so you know the background. In the past, we have used a generated mock for `nrf_modem_at_scanf` but it doesn't provide us all needed functionality due to variadic functions. Hence, we've had to make our own mock so we've kind of customized it. However, if 'custom' would mean that you take something as basis and then start modifying it, this is not custom but rather hand-written as you say. But the word custom also has some logic, because we could generate a mock for `nrf_modem_at_scanf` but as that's not enough, we have to customize it or write our own, whichever way you want to call it.

> That said, does this folder contain generic mocks that can be re-used by several tests, or are they `at` / `lib_modem` specific mocks that should go to a dedicated folder related to such tests.

This specific `nrf_modem_at` mock is generic and is used in this PR by tests/lib/location and tests/lib/lte_lc_api. There will be other users in the future as `nrf_modem_at` interface is a logical point for mocking in many tests. In SMS case, I actually had to write the code without using `nrf_modem_at_scanf` function because we didn't have a proper mock for it at the time `nrf_modem_at` interface had to be taken into use.

So the naming is in folder name, code file name (`mock_custom_nrf_modem_at.c`), nrf_modem_at Kconfig's and function names.
I do agree that the name of the folder should probably be just `mocks` irrespective how how nrf_modem_at konfigs and functions are named.
For Kconfig's and code file name just follow the function naming although maybe they don't need to.
There was separate conversation about function names and there is some level of preference to make a distinction between generated function name style and custom/hand-written mock's: https://github.com/nrfconnect/sdk-nrf/pull/8766#discussion_r976448300
I don't know what we should do with function naming as I do agree that some distinction here is good with the reasoning given in the other conversation.

Reply:
> This specific nrf_modem_at mock is generic and is used in this PR by tests/lib/location and tests/lib/lte_lc_api. 

Thanks for informing.

> There was separate conversation about function names and there is some level of preference to make a distinction between generated function name style and custom/hand-written mock's: https://github.com/nrfconnect/sdk-nrf/pull/8766#discussion_r976448300
I don't know what we should do with function naming as I do agree that some distinction here is good with the reasoning given in the other conversation.

I made a comment there, and I do agree the ability to distinguish is good to have, but I believe it to be better that code that is generated identifies itself as such, in this case CMock generated code ought to contain `cmock` or `generated` somewhere in its name (file / funciton), thereby leaving pure naming available to what we write by hand.

Reply:
Generated mocks are already distinguished by being prefixed with `__wrap`. I would propose that we instead agree to prefix the _non-generated_ mocks with something else, rather than extend the `__wrap` prefix to `__wrap_cmock` everywhere and then prefix the non-generated mocks with just `__wrap` or `__wrap_user` or whatever. This is a smaller change, it only affects one or two tests in this repo, as opposed to affecting everything (here and in other repos).

Plus, since there is no enforcing on the naming of hand-written mocks in either approach, I believe they are equivalent for our purposes, with the exception that one requires a lot of changes, and the other requires very little.